### PR TITLE
try to create target dir before saving token

### PR DESCRIPTION
**Pull request by struktured**
Using master, I had to manually create ```~/.opam/plugins/opam-publish/``` for the opam submit phase to complete. Otherwise I got an error like this:
```
Fatal error:
# opam-version    1.2.0
# os              linux
opam-publish: "open" failed on /home/carm/.opam/plugins/opam-publish/struktured.token: No such file or directory
```

This commit should fix the problem, although I couldn't fully verify it as I have nothing new to publish at this time.

### CODE REVIEW

File: src/opam_publish.ml [hunk 1]
```diff
@@ -139,6 +138,12 @@ let default_repo =
 let opam_publish_root =
   OpamFilename.OP.( opam_root / "plugins" / "opam-publish" )
 
+let create_opam_publish_root () =
+  try OpamFilename.mkdir opam_publish_root with
+  Unix.Unix_error (c, _, _) ->
+      OpamGlobals.warning "Error %s while creating opam publishing root"
```
Comment (dsheets) [inline src/opam_publish.ml]: Is `warning` strong enough for this error condition?

**Comment by dsheets** [comment]
I can confirm this works and the error was introduced in my patches which introduced secure token storage due to removing the use of the smart opam `write` function which creates any necessary directories. Thanks for the report, @struktured !

**Comment by struktured** [comment]
So a better solution here is to use the smart opam ```write``` function?

Regarding the message level when the directory is not created, I picked warn over something else in case the directory folder is readonly but the contents of the directory is not, which may mean the process can continue without error in the future.

**Comment by AltGr** [comment]
Thanks!

**Comment by dsheets** [comment]
I don't think the `write` function is a better solution because it does not give control over the permissions of the newly created file which contains security credentials.

I'm not sure that there is a 'safe' error allowing continued operation that can be raised from inside `OpamFilename.mkdir` <https://github.com/ocaml/opam/blob/master/src/core/opamSystem.ml#L61-L74>. Also, `Sys_error` <http://caml.inria.fr/pub/docs/manual-ocaml/core.html> is a possible exception.

**Comment by dbuenzli** [comment]
This PR eventually made me discover the `opam-publish` magic. I think the tool should be a little bit more verbose about it is doing and where it does it.

> I don't think the write function is a better solution because it does not give control over the permissions of the newly created file which contains security credentials.

Speaking of which, it seems however that the default permissions are somehow bad:
```
> ls -l .opam/plugins/opam-publish/dbuenzli.token 
-rw-r--r--  1 dbuenzli  staff  40 23 déc  2014 .opam/plugins/opam-publish/dbuenzli.token
```

**Comment by dsheets** [comment]
@dbuenzli, those were the old default permissions that were indeed bad.

If you are using the `master` branch (to be 0.3.0), new token files should be created with `600` and existing token files should be attenuated to `770 & st_perm`. See <https://github.com/OCamlPro/opam-publish/blob/master/src/opam_publish.ml#L304-L313>.

**Comment by dbuenzli** [comment]
I manually changed them. But it's good to know this is solved. Thanks.

**Pull request merged**